### PR TITLE
cpp-prefixsum

### PR DESCRIPTION
**Pull request by Beenishgul**

### CODE REVIEW

File: PIMbench/cpp-prefixsum/prefixsum.cpp [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+  exit(1);
+}
+  int value;
+  while (inputFile >> value) {
+  input.push_back(value);
+  }
+  inputFile.close();
+}
+
+void prefixSum(vector<int>& input, vector<int>& device_output, uint64_t len) {
```
[the remark this answers is not in the document]
Comment (mhgholamrezaei) [reply PIMbench/cpp-prefixsum/prefixsum.cpp:88]: @Beenishgul, could you please follow the same nameing convention as the other benchmark apps? I marked one of them.

**Review by mhgholamrezaei — CHANGES_REQUESTED** [review]

File: PIMbench/cpp-prefixsum/Makefile [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Makefile: C++ version of prefixsum
+# Copyright 2024 LavaLab @ University of Virginia. All rights reserved.
+
+include ../Makefile.common
+
+EXEC := prefixsum
```
Comment (fasiddique) [inline PIMbench/cpp-prefixsum/Makefile:6]:
```suggestion
EXEC := prefixsum.out
```

File: PIMbench/cpp-prefixsum/prefixsum.cpp [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+  uint64_t vectorLength;
+  char *configFile;
+  char *inputFile;
+  bool shouldVerify;
+} Params;
+
+void usage()
+{
+  fprintf(stderr,
+          "\nUsage:  ./prefixsum [options]"
```
Comment (fasiddique) [inline PIMbench/cpp-prefixsum/prefixsum.cpp:28]:
```suggestion
          "\nUsage:  ./prefixsum.out [options]"
```

File: PIMbench/cpp-prefixsum/prefixsum.cpp [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+struct Params getInputParams(int argc, char **argv)
+{
+  struct Params p;
+  p.vectorLength = 65536;
+  p.configFile = nullptr;
+  p.inputFile = nullptr;
+  p.shouldVerify = false;
+
+  int opt;
+while ((opt = getopt(argc, argv, "h:l:c:i:v:")) >= 0)
```
Comment (fasiddique) [inline PIMbench/cpp-prefixsum/prefixsum.cpp:46]: Indentation error

File: PIMbench/cpp-prefixsum/prefixsum.cpp [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+pimFree(tempObj);
+pimFree(inputObj);
+pimFree(outputObj);
+}
+
+int main(int argc, char* argv[]) {
+if (argc != 2) {
+    std::cerr << "Usage: " << argv[0] << " <number>\n";
+    return 1;
+}
```
Comment (fasiddique) [inline PIMbench/cpp-prefixsum/prefixsum.cpp:173]: This should be removed. Please check vector addition as a reference.

**Review by fasiddique — CHANGES_REQUESTED** [review]
The file is not properly indented. Please follow the other apps. A good example is vector addition.

**Review by mhgholamrezaei — APPROVED** [review]

**Review by fasiddique — APPROVED** [review]

**Pull request merged**